### DISCUSSION TITLE
Confused about the lense light profile and possible mask.

### DISCUSSION THREAD

**Discussion by seekermind** [AstroBridge/BDLensing#88] [Q&A]
Here is the image of the data.
![image](https://github.com/ajshajib/BDLensing/assets/48768846/b664f496-ee75-4083-8157-4a2638ebb0ed)

It's seems strange to me the vertical (cylindrical?) nature of the light profile. Also the arc is very faint at 4-6 o'clock. But again, there is no counter image. 
My initial masking guess is following.
![image](https://github.com/ajshajib/BDLensing/assets/48768846/ae0d204b-4431-41ac-9f17-c8417342c0d1)

Suggest me anything about masking region or any parameter.
Thanks.

**Comment by ajshajib** [comment]
The mask looks good for a trial fitting, so try to run the model fitting with it and show here what you get.

**Reply by seekermind** [reply]
I have run several mask region. But unfortunately none seems to even mildly match the system. Here are some of the run:

**Trial 1:**

_Mask:_ 
![image](https://github.com/ajshajib/BDLensing/assets/48768846/4484511b-1861-4d6b-88a8-84c1aae496d9)

_Params:_
`-2.3007570651132307 reduced X^2 of best position
-7992.830044203363 log likelihood
6948 effective number of data points
[{'theta_E': 2.158186808413816, 'gamma': 1.7916869774360928, 'e1': -0.08801346931946684, 'e2': 0.06447740503772757, 'center_x': -0.2249433780006856, 'center_y': -0.021383819113691842}, {'gamma1': -0.07781581961259795, 'gamma2': 0.05978981987557863, 'ra_0': -1.508734725566375, 'dec_0': -1.0043789364525657}] lens result
[{'amp': 1, 'R_sersic': 0.4718550023088072, 'n_sersic': 1.0, 'e1': 0.10915391169116755, 'e2': -0.12874519899512085, 'center_x': -0.3762312519680732, 'center_y': -0.3313359570483421}, {'amp': 1, 'n_max': 8, 'beta': 0.25104627850680467, 'center_x': -0.3762312519680732, 'center_y': -0.3313359570483421}] source result
[{'amp': 1, 'R_sersic': 0.9829775202631845, 'n_sersic': 4.0, 'e1': -0.08801346931946684, 'e2': 0.06447740503772757, 'center_x': -0.2249433780006856, 'center_y': -0.021383819113691842}, {'amp': 1, 'R_sersic': 0.19675102056693028, 'n_sersic': 1.0, 'e1': 0.06801626806983974, 'e2': 0.23411666648479507, 'center_x': 0.3853423988497692, 'center_y': 0.6908287751159644}] lens light result`

_Model figures:_

![image](https://github.com/ajshajib/BDLensing/assets/48768846/7ad9e379-dfe2-441d-bb42-8b6fdc179db6)
![image](https://github.com/ajshajib/BDLensing/assets/48768846/18684d95-ff16-48da-bb97-d2496682bae5)


**Trial 2:**


_Mask:_
![image](https://github.com/ajshajib/BDLensing/assets/48768846/bdadd27e-2e5f-4f75-8b9d-deee202dadd7)

_Params:_

`-1.4702826097764732 reduced X^2 of best position
-22634.265636203916 log likelihood
30789 effective number of data points
[{'theta_E': 2.000000135679467, 'gamma': 1.2181424596665564, 'e1': -0.1787705648017853, 'e2': 0.10756999028604229, 'center_x': -0.22902534800288185, 'center_y': -0.02365736577666136}, {'gamma1': -0.19584672686552357, 'gamma2': 0.10810670207144818, 'ra_0': -0.06108954000197239, 'dec_0': -0.03532699428586178}] lens result
[{'amp': 1, 'R_sersic': 0.48664642624069315, 'n_sersic': 1.0, 'e1': 0.4999636106934406, 'e2': -0.1356591642911419, 'center_x': -0.5675925219112751, 'center_y': 0.11107396388383864}, {'amp': 1, 'n_max': 8, 'beta': 0.2999990771625177, 'center_x': -0.5675925219112751, 'center_y': 0.11107396388383864}] source result
[{'amp': 1, 'R_sersic': 0.8090361301838459, 'n_sersic': 4.0, 'e1': -0.1787705648017853, 'e2': 0.10756999028604229, 'center_x': -0.22902534800288185, 'center_y': -0.02365736577666136}, {'amp': 1, 'R_sersic': 0.23705959765300996, 'n_sersic': 1.0, 'e1': -0.499996534243658, 'e2': 0.26004423394469434, 'center_x': -0.24149620741347919, 'center_y': -0.007469121062931856}] lens light result`

_Figures:_
![image](https://github.com/ajshajib/BDLensing/assets/48768846/2bd062d6-3b73-46b3-9cc3-ef103542e408)
![image](https://github.com/ajshajib/BDLensing/assets/48768846/408a6ef4-82c5-4d7c-8f12-6980fd782b7d)



**Trial 3:**

![image](https://github.com/ajshajib/BDLensing/assets/48768846/f0c93f71-c891-4512-876f-e6ba8f977a55)
![image](https://github.com/ajshajib/BDLensing/assets/48768846/0716801b-939a-4ca8-83a3-8597bac9161b)
![image](https://github.com/ajshajib/BDLensing/assets/48768846/cdf4baef-ec1b-418d-8a97-495a614bdac0)
![image](https://github.com/ajshajib/BDLensing/assets/48768846/bdd50ede-3540-40e5-b854-9f6f46aa309d)

**Reply by rafee-adnan** [reply]
[Here](https://drive.google.com/file/d/1m7duvInKE4JPqUlzdhno_N70KXugZBEM/view?usp=sharing) is the arc that I recreated. You can use these values to get a better fit.
![output](https://github.com/ajshajib/BDLensing/assets/76589056/127efbae-3f22-4b04-8bb0-77e958283e4c)
